controllers/storageautoscaling/storageautoscaling_controller.go

```diff
@@ -0,0 +1,107 @@
+package storageautoscaling
+
+import (
+	"context"
+	"crypto/tls"
+	"net/http"
+	"time"
+
+	"github.com/go-logr/logr"
+	prometheusapi "github.com/prometheus/client_golang/api"
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	prometheusconfig "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// StorageAutoscalerReconciler is the reconciler for StorageAutoscaler objects
+// nolint:revive
+type StorageAutoscalerReconciler struct {
+	client.Client
+	ctx               context.Context
+	Log               logr.Logger
+	Scheme            *runtime.Scheme
+	OperatorNamespace string
+	RequeueTime       time.Duration
+}
+
+// SetupWithManager sets up the reconciler with the manager
+func (r *StorageAutoscalerReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	storageAutoscalerReconcilerController := ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.Secret{}, builder.WithPredicates(predicate.GenerationChangedPredicate{}))
+		// todo: change configmap type to storageautoscaler
+
+	return storageAutoscalerReconcilerController.Complete(r)
+}
+
+// +kubebuilder:rbac:groups="monitoring.coreos.com",resources=*,verbs=get;list;watch
```

Comment:
@iamniting how can we set the `role`,
It is generating the `clusterrole`

Reply:
we get cluster roles only.